packages/react-native-codegen/src/parsers/flow/modules/index.js

```diff
@@ -249,39 +249,50 @@ function translateTypeAnnotation(
       }
     }
     case 'ObjectTypeAnnotation': {
-      const objectTypeAnnotation = {
-        type: 'ObjectTypeAnnotation',
-        // $FlowFixMe[missing-type-arg]
-        properties: ([
-          ...typeAnnotation.properties,
-          ...typeAnnotation.indexers,
-        ]: Array<$FlowFixMe>)
-          .map<?NamedShape<Nullable<NativeModuleBaseTypeAnnotation>>>(
-            property => {
-              return tryParse(() => {
-                return parseObjectProperty(
-                  property,
-                  hasteModuleName,
-                  types,
-                  aliasMap,
-                  tryParse,
-                  cxxOnly,
-                  nullable,
-                  translateTypeAnnotation,
-                  parser,
-                );
-              });
-            },
-          )
-          .filter(Boolean),
-      };
-
-      return typeAliasResolution(
-        typeAliasResolutionStatus,
-        objectTypeAnnotation,
-        aliasMap,
-        nullable,
-      );
+      // if there is any indexer, then it is a dictionary
+      if (
+        typeAnnotation.indexers &&
+        typeAnnotation.indexers.filter(
+          member => member.type === 'ObjectTypeIndexer',
+        ).length > 0
+      ) {
+        // no need to do further checking
+        return emitObject(nullable);
+      } else {
```

Comment:
probably we can completely remove the `else` and the extra indentation. The previous `if` branch immediately `return`. It should make it easier to read, what do you think?

Reply:
Fixed